components/esp_lvgl_port/esp_lvgl_port_fs.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dirent.h>
+#include <stdio.h>
+
+#include "esp_err.h"
+#include "esp_log.h"
+
+#include "freertos/FreeRTOS.h"
+
+#include "esp_lvgl_port.h"
+
+static const char *TAG = "lv_port_fs";
+
+static void fs_init(void);
+
+static void *fs_open (lv_fs_drv_t *drv, const char *path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close (lv_fs_drv_t *drv, void *file_p);
+static lv_fs_res_t fs_read (lv_fs_drv_t *drv, void *file_p, void *buf, uint32_t btr, uint32_t *br);
+static lv_fs_res_t fs_write(lv_fs_drv_t *drv, void *file_p, const void *buf, uint32_t btw, uint32_t *bw);
+static lv_fs_res_t fs_seek (lv_fs_drv_t *drv, void *file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_size (lv_fs_drv_t *drv, void *file_p, uint32_t *size_p);
+static lv_fs_res_t fs_tell (lv_fs_drv_t *drv, void *file_p, uint32_t *pos_p);
+
+static void *fs_dir_open (lv_fs_drv_t *drv, const char *path);
+static lv_fs_res_t fs_dir_read (lv_fs_drv_t *drv, void *rddir_p, char *fn);
+static lv_fs_res_t fs_dir_close (lv_fs_drv_t *drv, void *rddir_p);
+
+esp_err_t lvgl_port_fs_init(void)
+{
+    static bool lvgl_fs_inited = false;
+
+    if (lvgl_fs_inited) {
+        ESP_LOGI(TAG, "LVGL FS has been inited.");
+        return ESP_OK;
+    }
+
+    lvgl_fs_inited = true;
+
+    /*----------------------------------------------------
+     * Initialize your storage device and File System
+     * -------------------------------------------------*/
+    fs_init();
+
+    /*---------------------------------------------------
+     * Register the file system interface in LVGL
+     *--------------------------------------------------*/
+
+    /*Add a simple drive to open images*/
+    static lv_fs_drv_t fs_drv;
+    lv_fs_drv_init(&fs_drv);
+
+    /*Set up fields...*/
+    fs_drv.letter = 'S';
+    fs_drv.open_cb = fs_open;
+    fs_drv.close_cb = fs_close;
+    fs_drv.read_cb = fs_read;
+    fs_drv.write_cb = fs_write;
+    fs_drv.seek_cb = fs_seek;
+    fs_drv.tell_cb = fs_tell;
+
+    fs_drv.dir_close_cb = fs_dir_close;
+    fs_drv.dir_open_cb = fs_dir_open;
+    fs_drv.dir_read_cb = fs_dir_read;
+
+    lv_fs_drv_register(&fs_drv);
```

Comment:
Is it possible to use it as instance? For example rename this function to `lvgl_port_fs_add` and we can add more storages (e.g. SD card, SPIFS, ...).